packages/browser-data-collector/src/collectors/performance-marks.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import { getMarks } from '../api/user-timing';
+
+export const collector: Collector = ( report ) => {
+	const marks = getMarks();
+
+	// All data in the report is expected to be relative to `navigationStart`.
+	//
+	// mark.startTime is when the mark was craeted relative to the "time origin", which should
+	// be recorded in `performance.timeOrigin` but not all browsers supports it. However,
+	// `navigationStart` is reasonably close, therefore we don't need to correct `startTime`
+	marks.forEach( ( mark ) => {
```

Comment:
I can see a functional pattern in what we're doing for marks and measures here. Should we extract it to a helper function?